### ISSUE TITLE
[Theme]  I would expect the material components to respect the guidelines with the primary, lighter and darker colors

### ISSUE THREAD

**Issue by the-ult** [angular/material2#7679]
#### Bug, feature request, or proposal:
Proposal:
(This might be 2 proposals in one issue)

TL:DR;
-  mat-toolbar should default to the primary-dark hue
- `color` attribute should accept `primary`, `primary-light`, `primary-dark`, `accent`, `accent-light`, `accent-dark`, `warn`, `warn-light`, `warn-dark`
  - This will make it easier to follow the guidelines without having to override everything in (S)CSS

#### What is the expected behavior?
While defining a custom theme you can define 3 colors:
`$my-custom-theme: mat-palette($mat-blue, 500, 300, 900);`

I expect the material components to use the [default guideline colors](https://material.io/guidelines/style/color.html#color-color-system) (primary, light, dark) for the components.
And that it's easier to use the 3 different hue's

For example:
- `<mat-toolbar>My App</mat-toolbar>` => should default to the dark hue (900)
  - now defaults to no color
  - On the other hand, sometimes you'll need a transparent toolbar => maybe add `color="transparent"`
- `<mat-toolbar color="primary">My App</mat-toolbar>` => hue 500
- `<mat-toolbar color="primary-light">My App</mat-toolbar>` => hue 300
- `<mat-toolbar color="primary-dark">My App</mat-toolbar>` => hue 900

#### What is the current behavior?
- `mat-toolbar` does not follow the guidelines, while `mat-fab` does

atm I have to do this (right?)
```css
/* _base.material.scss =>global angular material overrides */
mat-toolbar {
  &.mat-toolbar {
    color: currentColor;
    &.mat-toolbar__header {
      &--darker {
        background-color: mat-color($bsm-primary, darker);
        color: mat-color($bsm-primary, default-contrast);
      }
  }
}
```
```html
<mat-toolbar class="mat-toolbar__header mat-toolbar__header--darker">
  My App
</mat-toolbar>
```
#### What is the use-case or motivation for changing an existing behavior?
- Easier to follow the material guidelines

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
```console
    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/
@angular/cli: 1.4.5
node: 8.1.4
os: win32 x64
@angular/animations: 4.4.4
@angular/cdk: 2.0.0-beta.12
@angular/common: 4.4.4
@angular/compiler: 4.4.4
@angular/core: 4.4.4
@angular/flex-layout: 2.0.0-rc.1
@angular/forms: 4.4.4
@angular/http: 4.4.4
@angular/material: 2.0.0-beta.12
@angular/platform-browser: 4.4.4
@angular/platform-browser-dynamic: 4.4.4
@angular/platform-server: 4.4.4
@angular/router: 4.4.4
@angular/cli: 1.4.5
@angular/compiler-cli: 4.4.4
@angular/language-service: 4.4.4
typescript: 2.5.3
```

#### Is there anything else we should know?

Love Angular and Angular Material 2. Keep up the great work!

**Comment by jelbourn** [comment]
We plan on adding support for this once we can drop support for IE11 and use css custom properties. Doing this with the current theming system would roughly triple the size of the theme css, so we're favoring smaller payload size here instead of adding this feature.

For toolbars, we aren't going to change the default color, but we could add a provider to let users _configure_ the default color. Feel free to file a new issue for the latter.
